lib/reply.js

```diff
@@ -312,19 +313,23 @@ Reply.prototype.code = function (code) {
 
 Reply.prototype.status = Reply.prototype.code
 
-Reply.prototype.getSerializationFunction = function (schemaOrStatus) {
+Reply.prototype.getSerializationFunction = function (schemaOrStatus, contentType) {
   let serialize
 
   if (typeof schemaOrStatus === 'string' || typeof schemaOrStatus === 'number') {
-    serialize = this[kRouteContext][kSchemaResponse]?.[schemaOrStatus]
+    if (contentType) {
```

Comment:
Let's just check the content type is a string, otherwise we just skip the lookup, wdyt?

Reply:
Since it should always be a string, it makes sense to be more strict, I will commit the change later today